### PR TITLE
CATROID-214 Add input to user defined bricks

### PR DESCRIPTION
**Pull request by mstoeg**
- Make User Brick scrollable so that it is possible to insert long inputs.
- Adapt Testcase AddUserBrickTest
- Create new classes for storing data (labels, inputs) of User Defined Bricks
- Create new class for storing User Defined Bricks and implement NEXT button for Inputs (described in CATROID-216)

https://jira.catrob.at/browse/CATROID-214

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/res/layout/brick_user_brick.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        android:gravity="center_vertical"
+        android:visibility="gone" />
+
+    <ScrollView
+        android:id="@+id/user_brick_scrollbar"
+        android:layout_marginTop="10dp"
+        android:layout_width="match_parent"
+        android:layout_height="@dimen/brick_height_medium"
+        android:layout_marginBottom="10dp">
+            <org.catrobat.catroid.ui.BrickLayout
```
Comment (BeinhAlex) [inline catroid/src/main/res/layout/brick_user_brick.xml:44]: Correction of intendation

File: catroid/src/main/java/org/catrobat/catroid/ui/BrickLayout.java [hunk 2]
```diff
@@ -178,7 +179,13 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
 			lineLength = lineLengthWithHorizontalSpacing + childWidth;
 			lineLengthWithHorizontalSpacing = lineLength + horizontalSpacing;
 
-			boolean newLine = (layoutParams.newLine && totalLengthOfContent - combinedLengthOfPreviousLines > sizeWidth);
+			boolean newLine;
+			if (userBrick) {
+				newLine = lineLengthWithHorizontalSpacing > sizeWidth;
+			} else {
+				newLine =
+						(layoutParams.newLine && totalLengthOfContent - combinedLengthOfPreviousLines > sizeWidth);
```
Comment (BeinhAlex) [inline catroid/src/main/java/org/catrobat/catroid/ui/BrickLayout.java:187]: Format change

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddInputToUserBrickFragment.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+		View view = inflater.inflate(R.layout.fragment_add_input_to_user_brick, container, false);
+
+		addInputUserBrickEditText = view.findViewById(R.id.input_user_brick_edit_field);
+		addInputUserBrickTextLayout = view.findViewById(R.id.input_user_brick_text_layout);
+		LinearLayout userBrickSpace = view.findViewById(R.id.user_brick_space);
+
+		Bundle arguments = getArguments();
+		if (arguments != null) {
+			userDefinedBrick =
```
Comment (BeinhAlex) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddInputToUserBrickFragment.java:80]: Format Change

File: catroid/src/main/res/menu/menu_user_brick.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (BeinhAlex) [inline catroid/src/main/res/menu/menu_user_brick.xml:1]: Reuse existing menu

File: catroid/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1052,12 +1052,18 @@
     <string name="category_assertions">Testing</string>
     <!--  -->
 
+
```
Comment (BeinhAlex) [inline catroid/src/main/res/values/strings.xml:1055]: Please remove multiple spaces

**Comment by wslany** [comment]
Thanks!

**Pull request merged**